packages/aragon-cli/src/commands/apm_cmds/publish.js

```diff
@@ -107,8 +107,50 @@ exports.builder = function(yargs) {
     })
 }
 
+const getMajor = version => version.split('.')[0]
+
+function getFunctionsAbi(functions, abi) {
+  functions.forEach(f => {
+    f.abi = abi.find(method => method.name === f.sig.match(/(.*)\(/)[1])
+  })
+}
+
+async function getDeprecatedFunctions(apm, artifact) {
+  let deprecated = {}
+  try {
+    let deprecatedOnVersion = []
+    const deprecatedFunctionsSig = []
+    const versions = await apm.getAllVersions(artifact.appName)
+    let lastMajor = -1
+    versions.reverse().forEach(async repo => {
+      // iterate on major versions
+      if (getMajor(repo.version) !== lastMajor) {
+        lastMajor = getMajor(repo.version)
+        repo.functions.forEach(f => {
```

Comment:
We should be careful here; if the IPFS content was lost for some reason, we wouldn't have the `repo.function` or `repo.abi`.

Maybe we could print a warning with this, and return early.